### DISCUSSION TITLE
Search by related fields

### DISCUSSION THREAD

**Discussion by arthurkirkosa** [binaryk/laravel-restify#396] [Ideas]
E.g.

We have `Invoices` that belong to a `Vendor`; the Vendor has `code` and `name` and we'd like to be able to use the global search endpoint to fully or partially search the Invoices repository and filter to only the Invoices that have Customers with the queried name or code

Just like with matches we should be able to define a structure that would auto "inject" the join in the main query when searching

**Comment by cristijora** [comment]
This would be awesome
